src/library/scala/collection/Map.scala

```diff
@@ -181,7 +181,12 @@ trait MapOps[K, +V, +CC[_, _] <: IterableOps[_, AnyConstr, _], +C]
 
   /** A set representing the keys contained by this map.
    *
-   *  For efficiency the resulting set may be a view (maintaining a reference to the map and reflecting modifications
+   *  By default, the set uses `keysIterator` for iteration and delegates `contains` to the `Map`.
+   *
+   *  For `Map` implementations with ordered but not sorted keys, such as a `SeqMap`,
+   *  `keys` may efficiently build a representation that preserves order.
```

Comment:
These two phrases tell me some implementation details and that maybe the order is preserved (but only maybe). I think this is potentially more confusing than helpful...

Reply:
I think `may` governs `efficiently`, not `preserves order`.